src/state/translations-mixin.ts

```diff
@@ -75,12 +83,41 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
       document.querySelector("html")!.setAttribute("lang", hass.language);
       this.style.direction = computeRTL(hass) ? "rtl" : "ltr";
       this._loadCoreTranslations(hass.language);
-      this._loadHassTranslations(hass.language);
+      this._loadHassTranslations(hass.language, "state");
       this._loadFragmentTranslations(hass.language, hass.panelUrl);
     }
 
-    private async _loadHassTranslations(language: string) {
-      const resources = await getHassTranslations(this.hass!, language);
+    private async _loadHassTranslations(
+      language: string,
+      category: Parameters<typeof getHassTranslations>[2],
+      integration?: Parameters<typeof getHassTranslations>[3],
+      configFlow?: Parameters<typeof getHassTranslations>[4],
+      force = false
+    ) {
+      // We don't cache individual integrations
+      // Our cache stores
+      if (
+        !force &&
+        !integration &&
+        // indicates if category is loaded
+        category in this.__loadedTranslations &&
+        // if this.__loadedTranslations[category] is true, it was loaded with config flow
+        (!configFlow || this.__loadedTranslations[category])
+      ) {
+        return;
+      }
+
+      if (!integration) {
+        this.__loadedTranslations[category] = configFlow === true;
```

Comment:
Can we make this clearer and more future proof?
```
        this.__loadedTranslations[category] = { configFlow };
```

Reply:
Also, if we will fetch the `title` translation of one page with, and the other without `configFlow` we keep busting the cache?